aspnet/mvc/views/working-with-forms.rst

```diff
@@ -0,0 +1,621 @@
+  :language: HTML
+
+The *Views/Shared/EditorTemplates/String.cshtml* template:
+
+.. literalinclude::   forms/sample/final/Views/Shared/EditorTemplates/String.cshtml 
+  :language: HTML
+  
+Sample using ``List<T>``:
+
+.. literalinclude::  forms/sample/final/ViewModels/ToDoItem.cs
+  :language: c#
+  :lines: 3-7
+  :dedent: 3
+
+The following Razor shows how to iterate over a collection:
+
+.. literalinclude::   forms/sample/final/Views/Demo/Edit.cshtml 
+  :language: HTML
+
+Using the *Views/Shared/EditorTemplates/ToDoItem.cshtml* template reduces evaluating the indexer from three to one.
+
+.. literalinclude::   forms/sample/final/Views/Shared/EditorTemplates/ToDoItem.cshtml 
+  :language: HTML
+  
+:Note: Always use ``for`` (and *not* ``foreach``) to iterate over a list. Evaluating an indexer in a LINQ expression can be expensive and should be minimized.
+:Note: The commented sample code above shows how you would replace the lambda expression with the ``@`` operator to access each ``ToDoItem`` in the list.
+
+HTML Helper alternative to Input Tag Helper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+``Html.TextBox()``, ``Html.TextBoxFor()``, ``Html.Editor()`` and ``Html.EditorFor()`` have overlapping features with the Input Tag Helper. The Input Tag Helper will automatically set the ``type`` attribute; ``Html.TextBoxFor()`` will not. ``Html.EditorFor()`` handles collections, complex objects and templates; the Input Tag Helper does not. The Input Tag Helper is strongly typed (it uses lambda expressions); ``Html.TextBox()`` and ``Html.Editor()`` are not (they use expression names).
+
+Navigating child properties 
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+You can also navigate to child properties of your view model. Consider a more complex model class that contains a child ``Address`` property.
+
+.. literalinclude::  forms/sample/final/ViewModels/AddressViewModel.cs
+  :language: c#
+  :lines: 5-8
+  :dedent: 3
+  :emphasize-lines: 1-
+
+.. literalinclude::  forms/sample/final/ViewModels/RegisterAddressViewModel.cs
+  :language: c#
+  :lines: 5-11
+  :dedent: 3
+  :emphasize-lines: 6
+
+In the view, we bind to ``Address.AddressLine1``: 
+
+.. literalinclude::  forms/sample/final/Views/Demo/RegisterAddress.cshtml 
+  :language: HTML
+  :emphasize-lines: 6
+
+The following HTML is generated for `` Address.AddressLine1``:
+
+.. code-block:: HTML
+
+  <input type="text" id="Address_AddressLine1" name="Address.AddressLine1" value="" />
+  
+The Textarea Tag Helper
+-------------------------
+
+The `Textarea Tag Helper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/TagHelpers/TextAreaTagHelper/index.html>`__ tag helper is  similar to the Input Tag Helper.  
+
+- Generates the ``id`` and ``name`` attributes, and the data validation attributes from the model for a `<textarea> <http://www.w3.org/wiki/HTML/Elements/textarea>`__ element. 
+- Provides strong typing. 
+- HTML Helper alternative: ``Html.TextAreaFor``
```

Comment:
References to HTML helper methods almost-always include parentheses up to this point. It's not consistent but seems to be a trend that's broken here. This document should use whichever of `Html.TextAreaFor` or `Html.TextAreaFor()` other documents in this repo use. The choice isn't important to me but removing the mix in this document is.